### PR TITLE
Add TestCategoryFilter to MSTestSettings issue #716

### PR DESCRIPTION
**Pull request by brettkc**
Add filtering option to MSTestSettings. Feedback welcome.

### CODE REVIEW

**Comment by brettkc** [comment]
Updated, I don't think the flags passed to MSTest need to be quoted (ex: "/noisolation"). The category filter itself needs to be quoted per the MSTest docs.

**Comment by gep13** [comment]
@brettkc looks like you have some StyleCop issues with this PR.  These are described here:

https://ci.appveyor.com/project/cakebuild/cake/build/0.9.0-PullRequest.717+71.build.1562#L776

Can you fix them up, so that we can continue the discussion?  Thanks!

**Comment by gep13** [comment]
**NOTE:** You should also see these errors when you run ".\build.ps1", or by switching into release mode within Visual Studio.

**Comment by brettkc** [comment]
@gep13 style issues should be resolved

**Comment by gep13** [comment]
@brettkc perfect, thanks for getting that fixed up.  The team will look at this as soon as possible, and get back to you with any comments.

File: src/Cake.Common/Tools/MSTest/MSTestSettings.cs [hunk 1]
```diff
@@ -18,6 +18,12 @@ public sealed class MSTestSettings : ToolSettings
         public bool NoIsolation { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating the test category filter string to pass to 
+        /// MSTest.exe flag /testcategory. See: https://msdn.microsoft.com/en-us/library/ms182489.aspx#category
+        /// </summary>
+        public string TestCategoryFilter { get; set; }
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSTest/MSTestSettings.cs]: I think this should be called just `Category` to better reflect the CLI switch.

**Comment by patriksvensson** [comment]
@brettkc Left a minor comment for you.

**Comment by brettkc** [comment]
@patriksvensson updated name of setting

**Comment by patriksvensson** [comment]
@brettkc Looks good to me! Could you rebase this against the develop branch and squash to a single commit? Thanks! :+1:

**Comment by brettkc** [comment]
@patriksvensson I think I've got the rebase correct..if you want to check it out. Thanks

**Comment by devlead** [comment]
@brettkc this branch is out of sync could you please [rebase](https://github.com/cake-build/cake/blob/develop/CONTRIBUTING.md#respond-to-feedback-on-pull-request) against latest develop and ping us when done.

**Comment by brettkc** [comment]
@devlead rebase done, not sure whats up on bitrise?

**Comment by devlead** [comment]
@brettkc merged :+1: Thanks for your contribution!

**Comment by gep13** [comment]
Relates to #716

**Pull request merged**